packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1417,6 +1417,84 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     return child;
   }
 
+  /// The [DelegatedTransition] provided to the route below this one in the
+  /// navigation stack.
+  ///
+  /// {@macro flutter.widgets.delegatedTransition}
+  ///
+  /// The [ModalRoute] receiving this transition will set it to their
+  /// [receivedTransition] property.
+  DelegatedTransition? get delegatedTransition => null;
+
+  /// The [DelegatedTransition] received from the route above this one in the
+  /// navigation stack.
+  ///
+  /// {@macro flutter.widgets.delegatedTransition}
+  ///
+  /// `receivedTransition` will use the above route's [delegatedTransition] in
+  /// order to show the right route transition when the above route either enters
+  /// or leaves the navigation stack. If not null, the `receivedTransition` will
+  /// wrap the route content through [buildFlexTransitions].
+  DelegatedTransition? receivedTransition;
```

Comment:
Can this be private?

Reply:
It maybe could be? But I feel like it's possible that a subclass might want to check it for controlling transition logic. Or overriding it for some reason. I don't have a solid reason to leave it public, but do too the nature of the PR enabling more customizability, I'm hesitant to make too many things private.

Reply:
I would only expose this if necessary, so that we have the freedom to change the implementation in the future without a breaking change.